numba/core/types/scalars.py

```diff
@@ -93,6 +93,14 @@ def can_convert_to(self, typingctx, other):
 
 
 Literal.ctor_map[int] = IntegerLiteral
+Literal.ctor_map[np.int8] = IntegerLiteral
+Literal.ctor_map[np.uint8] = IntegerLiteral
+Literal.ctor_map[np.int16] = IntegerLiteral
+Literal.ctor_map[np.uint16] = IntegerLiteral
+Literal.ctor_map[np.int32] = IntegerLiteral
+Literal.ctor_map[np.uint32] = IntegerLiteral
+Literal.ctor_map[np.int64] = IntegerLiteral
+Literal.ctor_map[np.uint64] = IntegerLiteral
```

Comment:
This is missing two types:
```suggestion
Literal.ctor_map[np.byte] = IntegerLiteral
Literal.ctor_map[np.ubyte] = IntegerLiteral
Literal.ctor_map[np.short] = IntegerLiteral
Literal.ctor_map[np.ushort] = IntegerLiteral
Literal.ctor_map[np.intc] = IntegerLiteral
Literal.ctor_map[np.uintc] = IntegerLiteral
Literal.ctor_map[np.int_] = IntegerLiteral
Literal.ctor_map[np.uint] = IntegerLiteral
Literal.ctor_map[np.longlong] = IntegerLiteral
Literal.ctor_map[np.ulonglong] = IntegerLiteral
```

Reply:
I was trying to decide what the right course of action was here - I used the fixed-size aliases because I thought it would be safest. 

At the moment I don't know much about literals in Numba and what the implication of using the canonical names is, I'm still trying to understand the context in which this is used.